### PR TITLE
Implement Weekend Payment Date Adjustment

### PR DESCRIPTION
**Pull request by sergeyklay**
### 🎯 Scope & Context
**Type:** Feat  
**Intent:** Add configurable weekend payment date adjustment so payment dates that fall on weekends can be resolved per-bill or via a global default to one of: unchanged, next_business_day, or previous_business_day. The change ensures displayed dates, AutoPay eligibility/processing, overdue detection, forecasts, settings, seeding, and APIs honor adjusted payment dates while preserving anchor-date recurrence semantics.

### 🧭 Reviewer Guide
**Complexity:** High

#### Entry Point
Start with `lib/services/DateAdjustmentService.ts` — the new pure utility implementing isWeekend, adjustPaymentDate, and getEffectiveStrategy which drives all adjustment behavior. Then review `db/schema.ts` and migration `drizzle/0013_melted_shotgun.sql` to see the new `weekend_adjustment` column and enum. Next inspect `lib/services/SettingsService.ts` (get/set global strategy) and `actions/settings.ts` (public actions and validation). Follow integrations in the core services:
- `lib/services/AutoPayService.ts` — eligibility now uses adjustedDueDate; transactions created with paidAt = adjustedDueDate; recurring bills advance by anchor date (prevents drift).
- `lib/services/RecurrenceService.ts` — daily overdue checks compare adjustedDueDate to today (includes TOCTOU considerations).
- `lib/services/ForecastService.ts` — projections compute anchor dates then apply adjustment for display and forecasting.
- `lib/services/BillService.ts` — new helper getAdjustedDueDate consolidating effective-strategy resolution.
Then verify API/UI and tooling surfaces:
- `actions/bills.ts` — `createBill` / `updateBill` accept and persist `weekendAdjustment` (nullable).
- `components/features/bills/BillFormDialog.tsx` — form schema, defaults, edit mapping, and select UI for per-bill override.
- `components/features/settings/WeekendAdjustmentDropdown.tsx` and `components/features/settings/SettingsSection.tsx` — settings UI and action wiring for global default.
- `scripts/seed-development.ts` — seeding updated to populate weekendAdjustment for scenario and template bills.
- `lib/types.ts`, `lib/constants.ts` — new `WeekendAdjustmentStrategy` type and `DEFAULT_WEEKEND_ADJUSTMENT`.
- Tests — extensive updates/additions across actions and services to cover adjustment logic and integrations (unit and integration-level tests).

#### Sensitive Areas
- `lib/services/AutoPayService.ts` - payment eligibility filters, transactional creation of transactions (`paidAt`) and recurrence advancement (monetary/processing correctness).  
- `lib/services/RecurrenceService.ts` - overdue determination and TOCTOU protections now use adjusted dates.  
- `lib/services/ForecastService.ts` - projection correctness (anchor date vs adjusted display date).  
- `lib/services/BillService.ts` - new public method `getAdjustedDueDate` and its callers.  
- `lib/services/SettingsService.ts` and `actions/settings.ts` - validation, persistence, and revalidation flows for global setting.  
- `db/schema.ts`, `drizzle/0013_melted_shotgun.sql`, `drizzle/meta/*` - schema change and migration snapshot.  
- `actions/bills.ts` and `components/features/bills/BillFormDialog.tsx` - handling of null (use global) vs explicit strategy values in API and UI.  
- `lib/services/DateAdjustmentService.ts` - date math edge cases (weekends, timezones, year boundaries, leap years).  
- Seed scripts and test fixtures - ensure seeded/default values align with `DEFAULT_WEEKEND_ADJUSTMENT`.

### ⚠️ Risk Assessment
- **Breaking Changes:** No breaking changes - `weekendAdjustment` is nullable/optional; code paths fall back to `DEFAULT_WEEKEND_ADJUSTMENT` ('unchanged') or per-bill override when present.  
- **Migrations/State:** Database migration required - `drizzle/0013_melted_shotgun.sql` adds the `weekend_adjustment` column and `drizzle/meta/0013_snapshot.json` / `_journal.json` updated. No other manual state changes required.

### CODE REVIEW

File: components/features/settings/WeekendAdjustmentDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      if (!result.success) {
+        toast.error('Failed to update setting', {
+          description: result.error || 'Please try again.',
+        });
+        setValue(currentValue);
+      } else {
+        toast.success('Setting updated');
+      }
+    });
+  };
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/settings/WeekendAdjustmentDropdown.tsx]: This one should be fixed now. Please recheck

File: components/features/bills/BillFormDialog.tsx [hunk 2]
```diff
@@ -175,6 +179,7 @@ export function BillFormDialog({
       const baseInput = {
         ...values,
         amount: parseMoneyInput(values.amount),
+        weekendAdjustment: values.weekendAdjustment ?? null,
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/bills/BillFormDialog.tsx]: This one should be fixed now. Please recheck

File: components/features/settings/useSettingDropdown.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  displayValue: T;
+  isPending: boolean;
+  handleValueChange: (newValue: string) => void;
+}
+
+export function useSettingDropdown<T extends string>({
+  currentValue,
+  onUpdate,
+  showSuccessToast = true,
+}: UseSettingDropdownOptions<T>): UseSettingDropdownReturn<T> {
```
[the remark this answers is not in the document]
Comment (sergeyklay) [reply components/features/settings/useSettingDropdown.ts]: This one is fixed. Please verify
